### PR TITLE
feat(Shared): Migrated View and Text Views

### PR DESCRIPTION
**Pull request by kevinvangelder**

### CODE REVIEW

File: ReactWindows/ReactNative.Net46/Views/Text/ReactTextViewManager.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media;
+
+namespace ReactNative.Views.Text
+{
+    /// <summary>
+    /// The view manager for text views.
+    /// </summary>
+    public class ReactTextViewManager : ViewParentManager<RichTextBox, ReactTextShadowNode>
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Views/Text/ReactTextViewManager.cs]: RichTextBox (.NET 4.6) == RichTextBlock (UWP)?

Comment (kevinvangelder) [inline ReactWindows/ReactNative.Net46/Views/Text/ReactTextViewManager.cs]: Close, from what I can tell. The UWP one mixes in some extra functionality and automatically manages the FlowDocument, where the WPF one is more manual.

File: ReactWindows/ReactNative.Shared/Views/Text/FontStyleHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        private static FontWeight ConvertNumeric(int fontWeightNumeric)
+        {
+#if WINDOWS_UWP
+            return new FontWeight
+                {
+                    Weight = (ushort)fontWeightNumeric,
+                };
+#else
+            if (fontWeightNumeric >= 0 && fontWeightNumeric <= 149) { return FontWeights.Thin; }
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Views/Text/FontStyleHelpers.cs]: >149 [](start = 63, length = 3)

probably would make more sense to just do `fontWeightNumeric < 150`, that way, if for whatever reason FontWeights moved from discrete to continuous, we'd be ready :-)

**Comment by rozele** [comment]
One thing that I don't particularly like is the "half-in, half-out" implementation of Text.  It's kind of a PITA when half of the files for a feature are in shared, and the rest are distributed across two projects.  I already had trouble with Animated yesterday (also half-in, half-out) because I wasn't sure where the right place to put a new AnimatedNode type was.

**Comment by kevinvangelder** [comment]
@rozele Do you want me to instead make complete copies in both projects? I can do that but that level of unnecessary redundancy is probably going to be more annoying than just putting cross-platform code in Shared and single-platform code in the applicable project.

Personally, I'd prefer to continue down this route, and if necessary, refactor it after our discussion during the hack week.

**Pull request merged**